tilavarauspalvelu/integrations/email/template_context/reservation.py

```diff
@@ -447,26 +704,82 @@ def get_context_for_reservation_requires_payment(
             begin_datetime=data["begin_datetime"],
             end_datetime=data["end_datetime"],
         ),
-        **get_contex_for_reservation_price(
-            price=data["price"],
-            tax_percentage=data["tax_percentage"],
-            booking_number=data["booking_number"],
-        ),
-        **get_contex_for_reservation_manage_link(language=language),
-        **get_contex_for_closing_polite(language=language),
+        **get_contex_for_seasonal_reservation_check_details_url(language=language),
+        **get_contex_for_closing(language=language),
     }
 
 
+# type: EmailType.SEASONAL_RESERVATION_REJECTED_SERIES #################################################################
+
+
 @overload
-def get_context_for_reservation_rejected(
-    reservation: Reservation,
+def get_context_for_seasonal_reservation_rejected_series(
+    reservation_series: RecurringReservation, *, language: Lang
+) -> EmailContext: ...
+
+
+@overload
+def get_context_for_seasonal_reservation_rejected_series(
     *,
     language: Lang,
+    rejection_reason: str,
+    email_recipient_name: str,
+    weekday_value: str,
+    time_value: str,
+    application_section_name: str,
+    application_round_name: str,
 ) -> EmailContext: ...
 
 
+@get_translated
+def get_context_for_seasonal_reservation_rejected_series(
+    reservation_series: RecurringReservation | None = None,
+    *,
+    language: Lang,
+    **data: Any,
+) -> EmailContext:
+    if reservation_series is not None:
+        section = reservation_series.actions.get_application_section()
+        reservation = reservation_series.reservations.first()
+
+        data: dict[str, Any] = {
+            "email_recipient_name": reservation_series.actions.get_email_reservee_name(),
+            "rejection_reason": get_attr_by_language(reservation.deny_reason, "reason", language),
+            **params_for_reservation_series_info(reservation_series=reservation_series),
+            **params_for_application_section_info(application_section=section, language=language),
+        }
+
+    return {
+        "title": pgettext("Email", "Your seasonal booking has been cancelled"),
+        "text_reservation_rejected": pgettext(
+            "Email", "All space reservations included in your seasonal booking have been cancelled"
```

Comment:
This seems to be different than in the defined email templates:
- **C** : staff cancels **one** series (possibly out of many) related to an application section: `"The space reservation included in your seasonal booking has been cancelled"` (same as emails A & B when a single reservation occurrence is cancelled/rejected but the weekday level time info clarifies how "reservation" is to be understood)
- **D** applicant cancels all series related to an application section: `"All space reservations included in your seasonal booking have been cancelled"`

Reply:
Nice catch!